src/LanguageServer.Common/Utilities/DotnetInfo.cs

```diff
@@ -0,0 +1,835 @@
+        ///     The exception message.
+        /// </param>
+        public DotnetDiscoveryException(string message)
+            : base(message)
+        {
+        }
+
+        /// <summary>
+        ///     Create a new <see cref="DotnetDiscoveryException"/>.
+        /// </summary>
+        /// <param name="message">
+        ///     The exception message.
+        /// </param>
+        /// <param name="innerException">
+        ///     The exception that caused the <see cref="DotnetDiscoveryException"/> to be raised.
+        /// </param>
+        public DotnetDiscoveryException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+    }
+
+    /// <summary>
+    ///     Information about a discovered version of the .NET SDK.
+    /// </summary>
+    /// <param name="Version">
+    ///     The SDK version.
+    /// </param>
+    /// <param name="BaseDirectory">
+    ///     The SDK base directory.
+    /// </param>
+    public record DotnetSdkInfo(SemanticVersion Version, string BaseDirectory)
+    {
+        /// <summary>
+        ///     Empty <see cref="DotnetRuntimeInfo"/>.
+        /// </summary>
+        public static readonly DotnetSdkInfo Empty = new DotnetSdkInfo(Version: null, BaseDirectory: null);
+    };
+
+    /// <summary>
+    ///     Information about a discovered .NET runtime.
+    /// </summary>
+    /// <param name="Name">
+    ///     The runtime name (e.g. "Microsoft.NETCore.App", "Microsoft.WindowsDesktop.App", "Microsoft.AspNetCore.App", etc).
+    /// </param>
+    /// <param name="Version">
+    ///     The SDK version.
+    /// </param>
+    public record DotnetRuntimeInfo(string Name, SemanticVersion Version)
+    {
+        /// <summary>
+        ///     Empty <see cref="DotnetRuntimeInfo"/>.
+        /// </summary>
+        public static readonly DotnetRuntimeInfo Empty = new DotnetRuntimeInfo(Name: null, Version: null);
+    };
+
+    /// <summary>
+    ///     Well-known names for various .NET runtimes.
+    /// </summary>
+    public static class WellKnownDotnetRuntimes
+    {
+        /// <summary>
+        ///     The name of the .NET runtime for ASP.NET.
+        /// </summary>
+        public static readonly string AspNetCore = "Microsoft.AspNetCore.App";
+
+        /// <summary>
+        ///     The name of the (default) .NET / .NET Core runtime.
+        /// </summary>
+        public static readonly string NetCore = "Microsoft.NETCore.App";
+
+        /// <summary>
+        ///     The name of the .NET runtime for Windows Desktop.
+        /// </summary>
+        public static readonly string WindowsDesktop = "Microsoft.WindowsDesktop.App";
+    }
+
+    /// <summary>
+    ///     Well-known process exit codes for the "dotnet" executable.
+    /// </summary>
+    /// <remarks>
+    ///     Derived from <seealso href="https://github.com/dotnet/runtime/blob/d123560a23078989f9563b83fa49a24802e88378/docs/design/features/host-error-codes.md"/>.
```

Comment:
```suggestion
    ///     Derived from <seealso href="https://github.com/dotnet/runtime/blob/main/docs/design/features/host-error-codes.md"/>.
```
Maybe make this always up-to-date?

Reply:
I've found that they sometimes move or rename doc pages and if you don't link to a specific commit, then the link eventually becomes broken :)